test/e2e/main_test.go

```diff
@@ -0,0 +1,16 @@
+package e2e
+
+import (
+	"os"
+	"testing"
+)
+
+func TestMain(m *testing.M) {
+	_, dirToCleanup := getProcessorArtifactPath()
+	os.RemoveAll(dirToCleanup)
```

Comment:
why clean both before and after?
can you extract this logic to a named function, `cleanNativeProcessorCache`?